iib/workers/tasks/build_merge_index_image.py

```diff
@@ -316,12 +317,32 @@ def handle_merge_request(
         set_request_state(request_id, 'in_progress', 'Deprecating bundles in the deprecation list')
         log.info('Deprecating bundles in the deprecation list')
         intermediate_bundles = missing_bundle_paths + source_index_bundles_pull_spec
+
         deprecation_bundles = get_bundles_from_deprecation_list(
             intermediate_bundles, deprecation_list
         )
+
         # We do not need to pass the invalid_version_bundles through the
         # get_bundles_from_deprecation_list function because we already know
         # they are present in the newly created index.
+
+        invalid_version_bundles_names = [
+            bundle["packageName"] for bundle in invalid_version_bundles
+        ]
+
+        filtered_invalid_version_bundles_names, _ = verify_operators_exists(
+            from_index=source_from_index_resolved,
```

Comment:
Maybe I told it to you wrongly. Now when I look to the code more deeply I am thinning about checking `intermediate_image_name` instead of `source_from_index_resolved`.
Because we run deprecate function on intermediate_image_name and this image should already contain data from source+target images.